known_invariants.md

```diff
@@ -28,16 +39,52 @@ var obj = {
 }
 ```
 
-Motivation: Developer Mental Model.
+### Specification Details : Which specifications are affected by this invariant
+
+TODO
+
+### Known Violations
+
+None
+
+### Rationale : Why did we have this invariant?
+
+Developer Mental Model.
 
 This invariant makes it seem as if both [[Get]] and [[Set]] are populated at the same time. It greatly simplifies how to think about objects and avoids forcing defensive programming against partially initialized objects.
 
 ## General Purpose Language
 
 The language is implementable independantly of the web platform.
 
-Motivation: ECMAScript is useful for many applications beyond the web, including on
-microcontrollers.
+### Specification Details : Which specifications are affected by this invariant
+
+TODO
+
+### Known Violations
+
+None
+
+### Rationale : Why did we have this invariant?
+
+ECMAScript is useful for many applications beyond the web, including on microcontrollers. By
+maintaining this invariant, we achieve the goal of maintaining a single language.
+
+## Prototype of (async) iterators
+
+When a built-in function creates an iterator, it should specify its prototype to be either %IteratorPrototype% or some %FooIteratorPrototype%.
```

Comment:
Invariants are “musts”, not shoulds, as i understand the term - iow, they’re supposed to be axioms, not merely ideals.

Reply:
I agree that the motivation needs more work. The issue is still open so we can discuss further there.